### PR TITLE
chore: Fix Some other issues faced during differential testing for 15 min window

### LINKED ISSUE #11250: Replay events from round 160916208 and validate record files for differential testing
Replay events from round 160916208 and validate record files for differential testing.

Found total 6 differences:
---- CryptoApproveAllowance DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Transaction fees '67757274' and '67757263' varied by more than 1 tinybar - Matching field 'transactionFee'  ==> expected: <true> but was: <false>

---- ContractCall DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched values, expected 'SUCCESS', got 'CONTRACT_REVERT_EXECUTED' - Matching field 'status' Matching field 'receipt'  ==> expected: <SUCCESS> but was: <CONTRACT_REVERT_EXECUTED>

---- ContractCall DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched values, expected 'SUCCESS', got 'REVERTED_SUCCESS' - Matching field 'status' Matching field 'receipt'  ==> expected: <SUCCESS> but was: <REVERTED_SUCCESS>

---- TokenMint DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched field counts  (expected but not find [newTotalSupply]) between expected status: SUCCESS

---- CryptoTransfer DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched field counts  (expected but not find [tokenTransferLists]) between expected receipt {

---- CryptoTransfer DIFFERED (TRANSACTION_MISMATCH) ----
Entries at position 61776 had different transactions (signedTransactionBytes: "\nA\n\026\n\f\b\314\237\343\255\006\020\274\257\237\250\002\022\004\030\200\354Q \004r\'\022%\n\005\030\215\242\310\001\022\r\n\004\030\373\342A\020\317\321\344\273\375\003\022\r\n\004\030\200\354Q\020\320\321\344\273\375\003"
 vs signedTransactionBytes: "\nC\n\026\n\f\b\314\237\343\255\006\020\274\257\237\250\002\022\004\030\200\354Q \004r)\022\'\n\005\030\215\242\310\001\022\017\n\004\030\373\342A\020\317\321\344\273\375\003\030\001\022\r\n\004\030\200\354Q\020\320\321\344\273\375\003"
)

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes https://github.com/hashgraph/hedera-services/issues/11205
Fixes #11250  

- When[ finalizing child record](https://github.com/hashgraph/hedera-services/blob/11205-15min-test/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java#L82), If the function is not a crypto transfer, then we filter all zero amounts from token transfer list. To be compatible with mono-service records, we _don't_ filter zero token transfers in the record. Same for the parent record finalization
- An empty transferList should be set only for a child record with status `SUCCESS`. But shouldn't be set when the status is `SUCCESS` to be compatible with mono-service
- Updated a fee issue to use `transactionValidStart` instead of `consensusTime` to [calculate lifeTime](https://github.com/hashgraph/hedera-services/blob/47202ad9a2ba4368d3eb81dd669cc330f37e0d31/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java#L526) for `cryptoApproveAllowanceHandler`

### CODE REVIEW

File: hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java [hunk 1]
```diff
@@ -73,9 +73,28 @@ public static List<DifferingEntries> findDifferencesBetweenV6(
...
         System.out.println(" ➡️  Read " + secondEntries.size() + " entries");
+        alignConsensusTimes(firstEntries, secondEntries);
         return diff(firstEntries, secondEntries, recordDiffSummarizer);
     }
 
+    private static void alignConsensusTimes(
+            final List<RecordStreamEntry> firstEntries, final List<RecordStreamEntry> secondEntries) {
+        int i = 0;
+        int j = 0;
+        // We assume that only second entries list will be missing entries
```
Comment (mhess-swl) [inline hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java:84]: Curious, why can we make this assumption?

Comment (Neeharika-Sompalli) [reply hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/forensics/OrderedComparison.java:84]: This isonly based on ethe evidence we saw until now in differential testing for 1 minute. There were always missed entries in mod-service generated records

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli !

**Review by iwsimon — APPROVED** [review]
LGTM, OutOfMemeoryError fixed by removed `alignConsensusTimes()`. Thanks @Neeharika-Sompalli

**Pull request merged**